tensorflow/compiler/mlir/xla/transforms/mhlo_to_lhlo_with_xla.cc

```diff
@@ -858,7 +865,11 @@ StatusOr<Operation*> LhloDialectEmitter::EmitCublasLtMatmul(
       custom_call->backend_config<xla::gpu::GemmBackendConfig>());
 
   bool has_matrix_bias = config.beta() != 0.;
-  bool has_vector_bias = config.epilogue() == xla::gpu::GemmBackendConfig::BIAS;
+
+  TF_ASSIGN_OR_RETURN(
+      bool has_vector_bias,
+      xla::gpu::cublas_lt::EpilogueAddsVectorBias(config.epilogue()));
```

Comment:
[tensorflow/compiler/mlir/xla/transforms/mhlo_to_lhlo_with_xla.cc:872:17: error: ‘xla::gpu::cublas_lt’ has not been declared

Where is this coming from?

Reply:
EpilogueAddsVectorBias is declared in [matmul_utils.h](https://github.com/tensorflow/tensorflow/blob/53726075eb6b39dcb0582e7f8fb9d417e8fe8950/tensorflow/compiler/xla/service/gpu/matmul_utils.h#L125), which is now [included](https://github.com/tensorflow/tensorflow/blob/53726075eb6b39dcb0582e7f8fb9d417e8fe8950/tensorflow/compiler/mlir/xla/transforms/mhlo_to_lhlo_with_xla.cc#L64) in mhlo_to_lhlo_with_xla.cc.

Reply:
EpilogueAddsVectorBias is protected by `#if GOOGLE_CUDA`. I wonder if that's the issue here.

Reply:
That's likely the cause. One possibility would be to keep the function in matmul_utils.h and move it outside the cublas_lt namespace.

Reply:
I think `#if GOOGLE_CUDA` is meant to wrap all code which depends on cuda libraries. `AsBlasLtEpilogue` is returning a value from `se::cuda::BlasLt::Epilogue` which is coming from _cuda_blas_lt.h_. All includes with this file seem to be guarded by this `#if`. I think the solution is more complex than you suggest.

Reply:
AsBlasLtEpilogue in gemm_algorithm_picker.cc could be moved to the other AsBlasLtEpilogue into the cublas_lt namespace.

Reply:
Ah sorry. My comment was not accurate and misleading.

I am trying to move `EpilogueAddsVectorBias` outside of `#if GOOGLE_CUDA` (but still within the `cublas_lt` namespace). I'll update with the results.